src/reflect/scala/reflect/internal/util/ScalaClassLoader.scala

```diff
@@ -129,10 +129,17 @@ object ScalaClassLoader {
 
   def setContext(cl: JClassLoader) = Thread.currentThread.setContextClassLoader(cl)
 
+  object URLClassLoader {
+    private val registered = new AtomicBoolean
+  }
   class URLClassLoader(urls: Seq[URL], parent: JClassLoader)
       extends JURLClassLoader(urls.toArray, parent)
          with ScalaClassLoader
          with HasClassPath {
+    // Note - this is calling a caller sensitive method so must be called from the class that is being registered
+    // and we are behaving like a java static method
+    if (URLClassLoader.registered.compareAndSet(false, true))
```

Comment:
```suggestion
    if (!URLClassLoader.registered.getAndSet(true))
```

Even better as it works better on those new Apple M1 silicon chips.